cpu/cc26x2_cc13x2/radio/rf_netdev.c

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright (C) 2020 Locha Inc
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_cc26x2_cc13x2
+ * @{
+ *
+ * @file
+ * @brief           CC13x2 IEEE 802.15.4 netdev driver
+ *
+ * @author          Jean Pierre Dudey <jeandudey@hotmail.com>
+ * @}
+ */
+
+#include <string.h>
+
+#include "cc26x2_cc13x2_rf.h"
+#include "cc26x2_cc13x2_rfc.h"
+#include "cc26xx_cc13xx_rfc_mailbox.h"
+#include "cc26xx_cc13xx_rfc_prop_mailbox.h"
+#include "cc26xx_cc13xx_rfc_queue.h"
+#include "vendor/rf_patch_cpe_prop.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#include "cpu.h"
+
+#define TX_BUF_SIZE (144)
+
+static uint8_t _tx_buf0[TX_BUF_SIZE] __attribute__ ((aligned(4)));
+
+#define RX_BUF_SIZE (144)
```

Comment:
Just a magic number to test the RX.

I've changed this to `IEEE802154G_FRAME_LEN_MAX` plus the bytes needed for the `rfc_data_entry_t` structure.

Might need #13979 to fully use the buffer (the 2047 bytes).

Reply:
Also for the `TX_BUF_SIZE` (and to remove magic numbers in `_send`) i've added the `IEEE802154_PHY_MR_FSK_PHR_LEN` (281ae3575454f383469a290442a611bf04c4e987) definition.

Reply:
Is this the same as `IEEE802154_FCS_LEN` or is that something else?

Reply:
PHR length is the length of the PHY header, they're different, however, the PHR specifies the FCS length, which can be 2 octects or 4.

![Screenshot from 2020-05-05 13-37-34](https://user-images.githubusercontent.com/5952531/81102700-9a87f100-8ed5-11ea-8ee6-cebc35e9a6e0.png)

Reply:
Finally I managed to change this to the size of the frame + the size of the data entry + 1 byte so the buffers in the array get aligned correctly, this needs to be modified (so we don't manually pad it) may be using an union for the RX buffer.

Something like this:

```c
typedef union {
    rfc_data_entry_t entry;
    uint8_t buf[IEEE802154G_FRAME_LEN_MAX + sizeof(rfc_data_entry_t)];
} rx_buf_t;
```

To avoid casting the bufer later to an `rfc_data_entry_t`.